docs/DeveloperGuide.md

```diff
@@ -152,15 +152,56 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 ## **Implementation**
 
+### Commands
+This section explains the general implementation of all commands.
+
+The following activity diagram generally shows the overall flow of events that the user will experience.
+
+
 This section describes some noteworthy details on how certain features are implemented.
 
+#### Parser Commands
+This section explains the implementation and execution of commands that have their own specific parser.
+
+Below is the sequence diagram for the execution of these commands (denoted by `XYZCommand`) after user input is sent to `LogicManager`. The execution of each of the command has been omitted due to their inherent differences and will be covered in their respective command sections below.
+
+![Command Parser Sequence Diagram](images/CommandsParserSequenceDiagram.png)
+
+Step 1:
+The user enters a command with the necessary parameters which is then passed to the `LogicManager`.
+
+Step 2:
+The `LogicManager` calls `AddressBookParser::parseCommand` for it to identify the type of command.
+
+Step 3:
+The `AddressBookParser` parses the user input and creates a command parser for that specific command. (denoted by `XYZCommandParser`)
+
+Step 4:
+The command parser is returned to the `AddressBookParser` which then calls `XYZCommandParser::parse` to parse the additional parameters.
+
+Step 5:
+The `XYZCommandParser` creates its respective command object (denoted by `XYZCommand`) and returns it to `LogicManager`.
+
+Step 6:
+The `LogicManager` calls `XYZCommand::execute` where the interaction between the command and the model is handled.
+
+Step 7:
+The `XYZCommand` creates a successful `CommandResult` and returns it to the UI.
+
+
+
 ### Ability to add persons
```

Comment:
It might be better to keep the title general then go into details for add_person with the diagrams you made